### ISSUE TITLE
Question: Is there a way to ignore YT shorts?

### ISSUE THREAD

**Issue by lue30499** [tubearchivist/tubearchivist#262]
As per the title.

**Comment by PhuriousGeorge** [comment]
Adding the filter `--match-filter 'original_url!*=/shorts/'` is supposed to ignore shorts in base yt-dlp.  I don't have a test environment setup, but don't think adding to the default download format entry would work.

**Comment by bbilly1** [comment]
It's tricky. Checkout out our previous discussion on the topic: #163, basically ytdlp doesn't detect that. The only feasible approach would be ignore by keyword, that will catch some if not most of the shorts. Its on the roadmap with:
> Auto ignore videos by keyword

**Comment by PhuriousGeorge** [comment]
> The only feasible approach would be ignore by keyword

The only issue with key_words_ only is if the word "short" or "#short" aren't used as tags, title or description.  I'm hoping yt-dlp comes up with something

Looking further into the issue linked from yt-dlp, appears they added checks for shorts to default to the /short/ url https://github.com/yt-dlp/yt-dlp/pull/3168, so the `--match-filter 'original_url!*=/shorts/'` filter should work?

**Comment by bbilly1** [comment]
Thanks for linking that. Looks like that was a recent change in yt-dlp. I've tested it with some videos, looks like the `original_url` is always set to the *shorts* link if you get a list of videos, for example: 
```bash
yt-dlp 'https://www.youtube.com/c/RealEngineering' --playlist-end 2 --dump-json
```
This returns two videos with original_url:
```
"original_url": "https://www.youtube.com/watch?v=8Oi8ZO-2Kvc"
...
"original_url": "https://www.youtube.com/shorts/br1p4fcqa4s"
```

But this doesn't work if you request the page with `watch?v=`, e.g. this:
```
yt-dlp 'https://www.youtube.com/watch?v=br1p4fcqa4s' --dump-json
```
will return 
```
"original_url": "https://www.youtube.com/watch?v=br1p4fcqa4s"
```
even though it is a shorts video. So yt-dlp can't detect it in this case. When Tube Archivist extracts channel videos for example, this extracts only ids first, then as a second step it that video needs to be added to the queue, will Tube Archivist download the full metadata. This significantly improves the rescan speed and reduces the requests to YouTube, so that's not something we should change just to be able to extract shorts.

TLDR: As long as yt-dlp can't detect shorts in the `watch?v=` url, the only approach I see would be keyword based.
